packages/split-button/src/Menu/Menu.tsx

```diff
@@ -0,0 +1,151 @@
+} from 'react';
+
+import Button from '@leafygreen-ui/button';
+import { cx } from '@leafygreen-ui/emotion';
+import { useBackdropClick, useEventListener } from '@leafygreen-ui/hooks';
+import Icon from '@leafygreen-ui/icon';
+import { useDarkMode } from '@leafygreen-ui/leafygreen-provider';
+import { isComponentType, keyMap } from '@leafygreen-ui/lib';
+import { Menu as LGMenu } from '@leafygreen-ui/menu';
+
+import { MenuItemType } from '../SplitButton';
+
+import {
+  triggerBaseStyles,
+  triggerSizeStyles,
+  triggerThemeStyles,
+} from './Menu.styles';
+import { MenuProps } from './Menu.types';
+
+/**
+ * @internal
+ */
+export const Menu = ({
+  variant,
+  disabled,
+  align,
+  justify,
+  size,
+  baseFontSize,
+  menuItems,
+  containerRef,
+  id,
+  onTriggerClick,
+  onChange,
+  triggerAriaLabel,
+  open: controlledOpen,
+  setOpen: controlledSetOpen,
+  ...rest
+}: MenuProps) => {
+  const { theme } = useDarkMode();
+  const [uncontrolledOpen, uncontrolledSetOpen] = useState<boolean>(false);
+  const buttonRef = useRef<null | HTMLButtonElement>(null);
+  const menuRef = useRef<null | HTMLDivElement>(null);
+
+  // TODO: make hook
+  const setOpen =
+    (typeof controlledOpen === 'boolean' && controlledSetOpen) ||
+    uncontrolledSetOpen;
+  const open = controlledOpen ?? uncontrolledOpen;
+
+  const handleTriggerClick: MouseEventHandler = e => {
+    onTriggerClick?.(e);
+    if (typeof controlledOpen !== 'boolean') {
+      setOpen(o => !o);
+    }
+  };
+
+  const handleClose = useCallback(() => {
+    buttonRef.current?.focus();
+    setOpen(false);
+  }, [setOpen]);
+
+  const handleKeyDown = (e: KeyboardEvent) => {
+    switch (e.keyCode) {
+      case keyMap.Tab:
+      case keyMap.Escape:
+        handleClose();
+        break;
+    }
+  };
+
+  useEventListener('keydown', handleKeyDown, { enabled: open });
+  useBackdropClick(handleClose, [buttonRef, menuRef], open);
+
+  const renderMenuItems = useMemo(() => {
+    const onMenuItemClick = (e: MouseEvent, menuItem: MenuItemType) => {
+      handleClose();
+      menuItem.props.onClick?.(e);
+      onChange?.(e);
+    };
+
+    const renderMenuItem = (menuItem: MenuItemType) => {
+      if (isComponentType(menuItem, 'MenuItem')) {
+        return React.cloneElement(menuItem, {
+          active: false,
+          key: `menuItem-${menuItem.key}`,
+          onClick: (e: MouseEvent) => onMenuItemClick(e, menuItem),
+        });
+      } else {
+        console.warn(
+          'Please use a LeafyGreen <MenuItem /> component. Received: ',
+          menuItem,
+        );
+      }
+    };
+
+    if (Array.isArray(menuItems) && menuItems.length) {
+      return menuItems.map(
+        (menuItem: MenuItemType): MenuItemType | undefined => {
+          if (menuItem == null) {
+            return menuItem;
+          }
+
+          return renderMenuItem(menuItem);
+        },
+      );
+    }
+  }, [handleClose, menuItems, onChange]);
+
+  return (
+    <>
+      <Button
```

Comment:
why not render this in the SplitButton file. That way you can share props between the two butttons and avoid passing all of these down

Reply:
It seemed a little neater to keep all the opening and closing logic inside of `Menu` as opposed to having all of that inside `SplitButton`  and passing some of that down to `Menu`.